applications/konami/src/module/konami_intercept.erl

```diff
@@ -0,0 +1,197 @@
+    lager:debug("update ~s to ~s with ctl ~s", [kapps_call:other_leg_call_id(Call), CallId, _ControlQueue]),
+
+    konami_code_fsm:transfer_to(
+      kapps_call:set_other_leg_call_id(CallId, Call)
+                               ,'b'
+     ),
+
+    {'stop', Call}.
+
+-spec source_leg_of_dtmf(ne_binary() | kz_json:object(), kapps_call:call()) -> 'a' | 'b'.
+source_leg_of_dtmf(<<_/binary>> = SourceDTMF, Call) ->
+    case kapps_call:call_id(Call) =:= SourceDTMF of
+        'true' -> 'a';
+        'false' -> 'b'
+    end;
+source_leg_of_dtmf(Data, Call) ->
+    source_leg_of_dtmf(kz_json:get_value(<<"dtmf_leg">>, Data), Call).
+
+-spec get_originate_req(kz_json:object(), kapps_call:call()) -> kz_proplist().
+get_originate_req(Data, Call) ->
+    SourceOfDTMF = kz_json:get_value(<<"dtmf_leg">>, Data),
+    TargetType = kz_json:get_value(<<"target_type">>, Data),
+    TargetId = kz_json:get_value(<<"target_id">>, Data),
+
+    Params = kz_json:set_values([{<<"source">>, ?MODULE}
+                                ,{<<"can_call_self">>, 'true'}
+                                ], Data),
+
+    SourceDeviceId = find_device_id_for_leg(SourceOfDTMF),
+    Endpoints = build_endpoints(TargetType, TargetId, SourceDeviceId, Params, Call),
+    build_originate(Endpoints, SourceOfDTMF, Call).
+
+-spec build_endpoints(ne_binary(), ne_binary(), ne_binary(), kz_json:object(), kapps_call:call()) -> kz_json:objects().
+build_endpoints(<<"device">>, Id, Id, _Params, _Call) ->
+    lager:debug("skipping ~s since it same source device id", [Id]),
+    [];
+build_endpoints(<<"device">>, DeviceId, _SourceDeviceId, Params, Call) ->
+    lager:debug("building endpoints for ~s", [DeviceId]),
+    case kz_endpoint:build(DeviceId, Params, Call) of
+        {'ok', Endpoint} -> Endpoint;
+        _Else -> []
+    end;
+build_endpoints(<<"user">>, OwnerId, SourceDeviceId, Params, Call) ->
+    lager:debug("building endpoints for user ~s", [OwnerId]),
+    lists:foldr(
+      fun(EndpointId, Acc) when EndpointId =:= SourceDeviceId ->
+              lager:debug("skipping ~s since it matches device id", [EndpointId]),
+              Acc;
+         (EndpointId, Acc) ->
+              lager:debug("building endpoint ~s", [EndpointId]),
+              case kz_endpoint:build(EndpointId, Params, Call) of
+                  {'ok', Endpoint} -> Endpoint ++ Acc;
+                  _Else -> Acc
+              end
+      end
+               ,[]
+               ,kz_attributes:owned_by(OwnerId, <<"device">>, Call)
+     ).
+
+-spec build_originate(kz_json:objects(), ne_binary(), kapps_call:call()) -> kz_proplist().
+build_originate([], _CallId, _Call) -> [];
+build_originate(Endpoints, CallId, Call) ->
+    lager:debug("targeting ~s for intercept", [CallId]),
+    props:filter_undefined(
+      [{<<"Application-Name">>, <<"bridge">>}
+      ,{<<"Endpoints">>, Endpoints}
+      ,{<<"Existing-Call-ID">>, CallId}
+      ,{<<"Intercept-Unbridged-Only">>, 'false'}
```

Comment:
@skorobkov this should be `true` for an `intercept` action or at least fetched from Data (cb_channels should get from params to build it)